src/components/ZonesPicker/index.js

```diff
@@ -0,0 +1,230 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Picker,
+  PickerSuggestions,
+  Input,
+  InputCheckbox,
+} from '..';
+import ZoneSuggestion from './ZoneSuggestion';
+import s from './ZonesPicker.css';
+import Suggestions from './ZonesPickerSuggestions';
+
+const getZoneSuggestionsFromValue = (zoneSuggestions, value) => zoneSuggestions.map((zone) => {
+  const ret = {
+    ...zone,
+    disabled: value.length > 0,
+  };
+  if (zone.name.toLowerCase() === value.toLowerCase()) {
+    ret.disabled = false;
+  }
+  return ret;
+});
+
+class ZonePicker extends React.PureComponent {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const {
+      fromZone: fromZoneProp,
+      toZone: toZoneProp,
+      fromZoneSuggestions,
+      toZoneSuggestions,
+    } = nextProps;
+    console.log('get dervied state', nextProps, prevState);
+
+    if (!fromZoneProp.length && !toZoneProp.length) {
+      return {
+        ...prevState,
+        fromZoneSuggestions:
+          getZoneSuggestionsFromValue(fromZoneSuggestions, prevState.fromZoneValue),
+        toZoneSuggestions: getZoneSuggestionsFromValue(toZoneSuggestions, prevState.toZoneValue),
+      };
+    }
+    const fromZone = fromZoneSuggestions.find(suggestion => suggestion.name === fromZoneProp);
+    const toZone = toZoneSuggestions.find(suggestion => suggestion.name === toZoneProp);
+
+    const newState = {
+      fromZoneValue: fromZone && fromZone.length >= prevState.fromZoneValue.length
+        ? fromZone
+        : prevState.fromZoneValue,
+      toZoneValue: toZone && toZone.length >= prevState.toZoneValue.length
+        ? toZone
+        : prevState.toZoneValue,
+    };
+    console.log('new state::', toZone, fromZone);
+    return {
+      ...newState,
+      fromZoneSuggestions: getZoneSuggestionsFromValue(fromZoneSuggestions, newState.fromZoneValue),
+      toZoneSuggestions: getZoneSuggestionsFromValue(toZoneSuggestions, newState.toZoneValue),
+    };
+  }
+
+  state = {
+    split: false,
+    fromZoneValue: '',
+    toZoneValue: '',
+    fromZoneSuggestions: [],
+    toZoneSuggestions: [],
+  };
+
+  constructor() {
+    super();
+
+    this.picker = React.createRef();
+  }
+
+  handleZoneClick = (zone, suggestionColumnPosition) => {
+    const { onSelect } = this.props;
+    const { split, fromZoneValue, fromZoneSuggestions } = this.state;
+    const newState = {};
+    let shouldClosePicker = false;
+
+    if (suggestionColumnPosition === 'first') {
```

Comment:
a mettre en constante en haut de ton composant ?

Reply:
Et ca serait pas mieux d'utiliser "from" pour first et "to" pour second parce que de ce que je comprends ca represente ca ?

Reply:
ces first et second viennent de la render prop inputcomponent du composant picker, je pense que c'est mieux ainsi parce que cette notion de from ou to est spécifique au picker de zone